tests/func/test_update.py

```diff
@@ -71,3 +73,33 @@ def test_update_import_url(tmp_dir, dvc, tmp_path_factory):
 
     assert dst.is_file()
     assert dst.read_text() == "updated file content"
+
+
+def test_update_rev(tmp_dir, dvc, erepo_dir, monkeypatch):
+    with monkeypatch.context() as m:
+        m.chdir(fspath(erepo_dir))
+        erepo_dir.scm.checkout("new_branch", create_new=True)
+        erepo_dir.dvc_gen("foo", "foo content", commit="create foo on branch")
+        erepo_dir.scm.checkout("master")
+        erepo_dir.scm.checkout("new_branch_2", create_new=True)
+        erepo_dir.dvc_gen(
+            "foo", "foo content 2", commit="create foo on branch"
+        )
+        erepo_dir.scm.checkout("master")
+
+    stage = dvc.imp(fspath(erepo_dir), "foo", "foo_imported", rev="new_branch")
+    dvc.update(stage.path, rev="new_branch_2")
+
+    assert (tmp_dir / "foo_imported").read_text() == "foo content 2"
+
+
+def test_update_rev_non_git_failure(repo_dir, dvc_repo):
```

Comment:
We are not using `repo_dir` and `dvc_repo` anymore, instead, we use `tmp_dir` or `dvc`.
This is not crucial, but it would be nice if you could read the docstring from `tests/dir_helpers.py`.
Sorry for the extra work, @ilgooz :disappointed:

Reply:
Here's a suggestion, to save you some time:

```python
def test_update_rev_non_git_failure(tmp_dir, dvc):
    tmp_dir.gen("file", "text")
    stage = dvc.imp_url("file", "imported")

    with pytest.raises(UpdateWithRevNotPossibleError):
        dvc_repo.update(stage.path, rev="dev")
```

Looks neat)